Sources/MarkupParser.swift

```diff
@@ -13,26 +13,27 @@ public struct MarkupParser {
 	}
 
 	private var tokenizer: MarkupTokenizer
-	private var openingDelimiters: [UnicodeScalar] = []
-
+	private var openingDelimiters: [(Int, UnicodeScalar)] = []
+	var index = 0
```

Comment:
What is this `index` var used for? You only increment it by 1 on line 32, but you don't read it anywhere else.

Reply:
updated logic